pkgs/development/libraries/json-glib/default.nix

```diff
@@ -14,25 +14,35 @@
 , gnome3
 }:
 
+let
+  isCross = stdenv.buildPlatform != stdenv.targetPlatform;
+in
 stdenv.mkDerivation rec {
   pname = "json-glib";
   version = "1.6.2";
 
-  outputs = [ "out" "dev" "devdoc" ];
+  outputs = [ "out" "dev" ] ++ lib.optionals (!isCross) [ "devdoc" ];
 
   src = fetchurl {
     url = "mirror://gnome/sources/${pname}/${lib.versions.majorMinor version}/${pname}-${version}.tar.xz";
     sha256 = "092g2dyy1hhl0ix9kp33wcab0pg1qicnsv0cj5ms9g9qs336cgd3";
   };
 
+  strictDeps = true;
+
+  depsBuildBuild = [
+    pkg-config
+  ] ++ lib.optionals (!isCross) [
+    gobject-introspection
```

Comment:
GIR files produced by `gobject-introspection` are platform specific (they basically describe the ABI of the library) so IIUC `gobject-introspection` should target host platform and therefore should go to `depsBuildHost` (i.e. `nativeBuildInputs`).